vignettes/adtr.Rmd

```diff
@@ -0,0 +1,599 @@
+required input to downstream parameter derivations.  See examples below.
+
+### Partial Date Imputation and Deriving `ADT`, `ADTF`, `AVISIT` etc
+
+If your data collection allows for partial dates, you could apply a
+company-specific imputation rule at this stage when deriving `ADT`. For this
+example, here we impute missing day to earliest possible date.
+
+```{r}
+tr <- derive_vars_dt(
+  tr,
+  dtc = TRDTC,
+  new_vars_prefix = "A",
+  highest_imputation = "D",
+  date_imputation = "first"
+) %>%
+  derive_vars_dy(
+    reference_date = RANDDT,
+    source_vars = vars(ADT)
+  )
+```
+
+
+```{r, echo=FALSE}
+dataset_vignette(tr, display_vars = vars(USUBJID, RANDDT, TRLNKID, TRDTC, ADT, ADTF, ADY))
+```  
+
+### Re-labelling unscheduled visits
+
+If you have data collected at unscheduled visits, which are not identified as
+unique in your SDTM or you have company specific rules.  Below is an example of
+how it could be re-labelled. <add more information once I see the template code>
+
+```{r}
+tr <- mutate(
+  tr,
+  AVISIT = VISIT,
+  AVISITN = VISITNUM
+)
+```
+
+## Derive Parameters TLDn and TLPn {#parameters}
+
+As mentioned earlier, depending on your company specifications you may either
+create parameters for TLDn and TLPn in ADTR or similar dataset.
+
+Set the `AVALx/ANL01FL` variables.
+```{r}
+adtr <- tr %>%
+  mutate(
+    tmp_lesion_nr = str_sub(TRLNKID, 3),
+    PARAMCD = paste0("DIAM", tmp_lesion_nr),
+    PARAM = paste("Target Lesion", tmp_lesion_nr, "Analysis Diameter by Investigator"),
+    PARCAT1 = "Target Lesion(s)",
+    PARCAT2 = "Investigator",
+    PARCAT3 = "Recist 1.1",
+    AVAL = TRSTRESN,
+    ANL01FL = if_else(!is.na(AVAL), "Y", NA_character_)
+  ) %>%
+  select(-tmp_lesion_nr)
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  arrange(adtr, USUBJID, TRLNKID, AVISITN),
+  display_vars = vars(USUBJID, TRLNKID, AVISIT, PARAMCD, PARAM, AVAL, ANL01FL)
+)
+```
+
+## Derive Parameters (TSD) and Analysis Flag `ANL01FL` {#tsd}
+
+Target Lesions Sum of Diameters is the sum of the diameters for lymph and
+non-lymph nodes (sum of analysis values `AVAL` from source observations TLDn &
+TLPn).  The analysis flag `ANL01FL` flags where the number of lesions assessed
+at baseline and at post-baseline match.  To assess whether the number of lesions
+expected and assessed match, you could compare lesion identifiers (`TR.TRLNKID`)
+/use the previous variables set up earlier to compare lesion expected vs lesion
+accessed  or check for target lesion response (from `RS`) =NE.
+
+```{r}
+adtr_sum <- get_summary_records(
+  adtr,
+  by_vars = vars(STUDYID, USUBJID, !!!adsl_vars, AVISIT, AVISITN),
+  analysis_var = AVAL,
+  summary_fun = function(x) sum(x, na.rm = TRUE),
+  set_values_to = vars(
+    PARAMCD = "TSD",
+    PARAM = "Target Lesions Sum of Diameters by Investigator",
+    PARCAT1 = "Target Lesion(s)",
+    PARCAT2 = "Investigator",
+    PARCAT3 = "Recist 1.1"
+  )
+)
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
```

Comment:
Is it me or does this `dataset_vignette`  display slow the RMD down?

Reply:
That's a known issue. We should restrict the example data to a few patients.